source/numeric/SVInt.cpp

```diff
@@ -1059,24 +1059,38 @@ SVInt& SVInt::operator&=(const SVInt& rhs) {
             return *this &= rhs.extend(bitWidth, bothSigned);
     }
 
+    if (!hasUnknown() && rhs.hasUnknown())
+        makeUnknown();
     if (isSingleWord())
         val &= rhs.val;
     else {
         uint32_t words = getNumWords(bitWidth, false);
         if (unknownFlag) {
-            for (uint32_t i = 0; i < words; i++)
-                pVal[i + words] = (pVal[i + words] | rhs.pVal[i + words]) &
-                                  (pVal[i + words] | pVal[i]) & (rhs.pVal[i + words] | rhs.pVal[i]);
+            if (rhs.isSingleWord()) {
+                pVal[1] &= rhs.val;
+                pVal[0] = ~pVal[1] & pVal[0] & rhs.val;
+            }
+            else {
+                if (rhs.hasUnknown())
+                    for (uint32_t i = 0; i < words; i++)
```

Comment:
Can you please use braces around any if or for body that spans more than one line, even if it is logically just a single statement.

Reply:
Yes, for example, the braces are inserted for
```c++
                if (rhs.hasUnknown()) {
                    for (uint32_t i = 0; i < words; i++) {
                        pVal[i + words] = (pVal[i + words] | rhs.pVal[i + words]) &
                                          (pVal[i + words] | pVal[i]) &
                                          (rhs.pVal[i + words] | rhs.pVal[i]);
                    }
                }
                else {
                    for (uint32_t i = 0; i < words; i++)
                        pVal[i + words] &= rhs.pVal[i];
                }
```